staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -242,7 +244,18 @@ func (o *WaitOptions) RunWait() error {
 			return fmt.Errorf("%v unsatisified for unknown reason", finalObject)
 		}
 		return err
-	})
+	}
+	visitor := o.ResourceFinder.Do()
+	if result, ok := visitor.(*resource.Result); ok && strings.ToLower(o.ForCondition) == "delete" {
+		result.IgnoreErrors(apierrors.IsNotFound)
```

Comment:
Yeah, this looks reasonable. Can you drop the rest, though, since it's identical with the rest of this function.

Reply:
Done :)